_posts/2016-01-20-why-try-java-8-functional-programming.markdown

```diff
@@ -0,0 +1,133 @@
+---
+layout: post
+comments: true
+date: 2016-01-20 08:13:00+0100
+authors: Sjur Millidahl
+title: "Why Try - Java 8 lambdas and checked Exceptions"
+tags:
+- java
+- java8
+- functional
+- lambda
+- exceptions
+- monad
+- try
+- success
+- failure
+---
+
+<figure>
+  <img src="/images/2016-01-20-why-try-java-8-functional-programming/thumbnail.png" alt="lambdas dislike checked exceptions" />
+  <figcaption>Why won't this compile? (And whyyy does riskyTask throw a PrinterException?!)</figcaption>
+</figure>
+
+One does not have to use lambdas in Java 8 long before running into the obstacle of checked Exceptions. Because the PrinterException is checked, the compiler forces us to deal with it, even within a lambda:
+
+{% highlight java %}
+private void demonstrate() {
+    IntStream ones = IntStream.generate(() -> 1);
+    ones.map(i -> {
+        try {
+            return riskyTask(i);
+        } catch (PrinterException e) {
+            System.out.println(
+                "Your printer is out of ink or laser beams!"
+            );
+        }
+        return i;
+    });
+}
+
+private int riskyTask(int a) throws PrinterException {
+    return 2;
+}
+{% endhighlight %}
+
+But we don't like to do this. We use lambdas to express intent in a concise and elegant fashion. The try/catch-brackets feels like noise.
+
+For this reason, FINN.no's open source  [lambda-companion](https://github.com/finn-no/lambda-companion) project introduces a useful structure for using lambdas in a world with checked Exceptions : Try.
+
+A Try represents a computation which might fail, and is always represented as a Success or a Failure (but never both). The concept borrows from Scala's Try, and shares several properties to other monadic functional structures :
```

Comment:
"Scala's Try" => make this a link to doc (http://www.scala-lang.org/api/current/index.html#scala.util.Try)?